src/StringUtils.cpp

```diff
@@ -140,6 +140,48 @@ AStringVector StringSplit(const AString & str, const AString & delim)
 
 
 
+AStringVector StringSplitWithQuotes(const AString & str, const AString & delim)
+{
+	AStringVector results;
+
+	size_t cutAt = 0;
+	size_t Prev = 0;
+
+	while ((cutAt = str.find_first_of(delim, Prev)) != str.npos)
+	{
+		AString current = str.substr(Prev, cutAt - Prev);
+		if ((current.at(0) == '"') || (current.at(0) == '\''))
+		{
+			Prev += 1;
+			cutAt = str.find_first_of(current.at(0), Prev);
+			if (cutAt != str.npos)
+			{
+				current = str.substr(Prev, cutAt - Prev);
+				cutAt += 1;
+			}
+		}
+
+		results.push_back(current);
+		Prev = cutAt + 1;
+	}
+
+	if (Prev < str.length())
+	{
+		AString current = str.substr(Prev);
+
+		if ((current.length() >= 2) && ((current.front() == '"') || (current.front() == '\'')) && (current.front() == current.back()))
```

Comment:
This condition should be probably split onto multiple lines, along the `&&` 'separators'. Also a comment would be nice:
```cpp
// If the remnant is wrapped in matching quotes, remove them:
if (
  (current.length() >= 2) && 
  ((current.front() == '"') || (current.front() == '\'')) && 
  (current.front() == current.back())
)
{
  ...
```